### ISSUE TITLE
Setup failing on multi-site setup

### ISSUE THREAD

**Issue by shoored** [dyerc/craft-flux#3]
Hello,

After installing and filling in the basic setup details like S3 bucket, Cloudfront ID etc I then proceeded to test the connection.

This resulted in the following error:

Unfortunately I cannot view the plugin's settings anymore. 
Please do let me know if you need more details.

```
Twig\Error\RuntimeError: Variable "siteOptions" does not exist. in /var/www/html/vendor/dyerc/craft-flux/src/templates/_settings.twig:23
```

```
1. in /var/www/html/vendor/dyerc/craft-flux/src/templates/_settings.twig at line 23
14151617181920212223242526272829303132  general: {label: 'General Settings'|t('flux'), url: '#general'},
  aws: {label: 'AWS Settings'|t('flux'), url: '#aws'},
  advanced: {label: 'Advanced Settings'|t('flux'), url: '#advanced'},
} %}
 
{% set fullPageForm = true %}
 
{% set siteColumn = [] %}
{% if craft.app.getIsMultiSite() %}
  {% set allSiteOptions = [{value: '', label: 'All Sites'|t('flux')}]|merge(siteOptions) %}
  {% set siteColumn = {
    siteId: {
      type: 'select',
      heading: 'Site'|t('flux'),
      options: allSiteOptions,
      thin: true,
    }
  } %}
{% endif %}
```

![image](https://user-images.githubusercontent.com/26165948/214270745-757e9fb3-73f6-4295-9c08-77599a8317bb.png)

**Comment by shoored** [comment]
Update: I just saw a failed task "Updating AWS Flux Installation". Reran it and this is the output:

```
Error executing "CreateRole" on "https://iam.amazonaws.com"; 

AWS HTTP error: Client error: `POST https://iam.amazonaws.com` resulted in a `403 Forbidden` response: 
<ErrorResponse xmlns="https://iam.amazonaws.com/doc/2010-05-08/"> <Error> <Type>Sender</Type> 
<Code>AccessDeni (truncated...) AccessDenied (client): 
User: arn:aws:iam::374092736628:user/xxxxx is not authorized to perform: iam:CreateRole on resource: 
arn:aws:iam::374092736628:role/xxxxx-Viewer-Request-Role because no identity-based policy allows the iam:CreateRole 
action - <ErrorResponse xmlns="https://iam.amazonaws.com/doc/2010-05-08/"> <Error> <Type>Sender</Type> 
<Code>AccessDenied</Code> <Message>User: arn:aws:iam::374092736628:user/xxxxx is not authorized to perform: 
iam:CreateRole on resource: arn:aws:iam::374092736628:role/xxxxx-Viewer-Request-Role because no identity-based policy 
allows the iam:CreateRole action</Message> </Error> <RequestId>891d3b6d-0785-4086-9e88-188bb0ed1143</RequestId> </ErrorResponse>
```

Will investigate some more.

**Update**
I was able to disable the plugin, re-enable it and redo the basic configuration along with creating a new IAM user and attaching the provided policy.
Also the "Updating AWS Flux Installation" task has been completed. 
So the only remaining issue at this stage is that I cannot access the plugin's settings because of the error mentioned above.

**Comment by dyerc** [comment]
Hi @shoored,

Sorry to hear you've encountered this issue, thanks for letting me know. If you run `composer update` with the following it should pull in my fix:

```
"require": {
   "dyerc/craft-flux": "dev-develop",
}
```

The error in the failed task will probably still persist, but you should now be able to see the settings screens correctly. Let me know how you get on?

**Comment by shoored** [comment]
Thanks for the quick response @dyerc, that resolves the issue! 

Getting some different cloudfront errors on the front-end now but that's a different story.

**Comment by dyerc** [comment]
👍  great, I'll close this issue and deploy the fix in version 4.1.2. Please drop me an email or open another issue if you encounter any problems.
